basilica/basilica/__init__.py

```diff
@@ -81,16 +86,60 @@ def raw_embed(self, url, data, opts, timeout):
     # TODO: parallelize
     def embed(self, url, data, batch_size, opts, timeout):
         batch = []
+        outstanding = 0
+        api_thread = threading.Thread(target=self.raw_embed_wrapper, args=(url, opts, timeout))
+        api_thread.setDaemon(True)
+        api_thread.start()
         for i in data:
             batch.append(i)
             if len(batch) >= batch_size:
-                for e in self.raw_embed(url, batch, opts=opts, timeout=timeout):
-                    yield e
+                self.write_queue.put_nowait(batch)
                 batch = []
+                if self.thread_running:
+                    emb = self.read_queue.get(block=True)
+                    if isinstance(emb, Exception):
+                        raise emb
+                    for e in emb:
+                        yield e
+                try:
+                    emb = self.read_queue.get(block=False) 
+                except Queue.Empty:
+                    continue
+                else:
+                    for e in emb:
+                        yield e
         if len(batch) > 0:
-            for e in self.raw_embed(url, batch, opts=opts, timeout=timeout):
-                yield e
-            batch = []
+            self.write_queue.put_nowait(batch)
+        while not self.write_queue.empty() or self.thread_running:
+            try:
+                emb = self.read_queue.get(block=True)
+                if isinstance(emb, Exception):
+                    raise emb
+            except Queue.Empty:
+                continue
```

Comment:
Is it possible to get a `Queue.Empty` exception when calling `get` with `block=True`?